### ISSUE TITLE
genTransactionNotFoundResponse

### ISSUE THREAD

**Issue by xboston** [metahashorg/crypt_example_c#1]
Приветствую!

Согласно спецификации JSON-RPC формат ответа сообщения ошибки должен располлогать расширенные информационные поля в блоке `data`, сейчас в корневом поле `error` есть лишние поля `countBlocks` и `knwonBlock`, что ломает RPC-клиенты при запросе несуществующей транзакции, ну и само поле `knwonBlock` с ошибкой написано:

```
{
  "error": {
    "code": -32603,
    "message": "Transaction 66f665fbf0b2f9971bca9ac45d86bbe11369b51e234f149a0e79e9bab152f03d not found",
    "countBlocks": 137910,
    "knwonBlock": 137910
  }
}
```

Всё это тут https://github.com/metahashorg/metahash-fullnode-client/blob/master/src/generate_json.cpp#L45

PR не стал делать, думаю такой формат в core-нодах присутствует и лучше сразу везде исправлять.

**Comment by Andreano25** [comment]
Добрый день.

Да действительно есть опечатка.
По поводу json-rpc, в спецификации (https://www.jsonrpc.org/specification) написано:

> data
>     A Primitive or Structured value that contains additional information about the error.
>     **This may be omitted.**
>     The value of this member is defined by the Server (e.g. detailed error information, nested errors etc.).

Примеры в этой же спецификации показывают, что это поле не обязательное.

**Comment by xboston** [comment]
@Andreano25 да, поле `data` не обязательно, но именно в нём должны быть размещены эти самые `additional information`, в нашем случае ответ должен быть таким:

```json
{
  "error": {
    "code": -32603,
    "message": "Transaction 66f665fbf0b2f9971bca9ac45d86bbe11369b51e234f149a0e79e9bab152f03d not found",
    "data": {
        "countBlocks": 137910,
        "knownBlock": 137910
    }
  }
}
```
